File: changsukuaidi-express-andriod/app/src/main/java/com/changsukuaidi/www/modules/home/HomeModule.java
package com.changsukuaidi.www.modules.home;

import dagger.Module;
import dagger.Provides;

/**
 * @Describe
 * @Author zhouhao
 * @Date 2018/2/12
 * @Contact 605626708@qq.com
 */

@Module
class HomeModule {

    private final IHomeView mView;

    HomeModule(IHomeView view) {
        mView = view;
    }

    @Provides
    IHomeView provideHomeView(){
        return mView;
    }
}

File: WIFIDirect/app/src/main/java/com/thc/www/p2p/module/HomeModule.kt
package com.thc.www.p2p.module

import com.thc.www.p2p.android.home.HomeFragment
import com.thc.www.p2p.annotation.FragmentScope
import dagger.Module
import dagger.Provides

/**
 * @Describe
 * @Author thc
 * @Date  2018/5/7
 */
@Module
class HomeModule(private val view: HomeFragment) {
    @FragmentScope
    @Provides
    fun provideView() = view
}